frontend/catalyst/utils/filesystem.py

```diff
@@ -66,7 +67,10 @@ class TemporaryDirectorySilent(tempfile.TemporaryDirectory):
     def _cleanup(cls, name, warn_message, **kwargs):  # pylint: disable=arguments-differ
         """Ignore ResourceWarning during cleanup."""
         del warn_message
-
+        minor_version = sys.version_info[1]
+        if kwargs.get("delete") and minor_version >= 12:
+            # Changed in version 3.12: Removed the "delete" kwargs
+            del kwargs["delete"]  # pragma: nocover
```

Comment:
Oh thanks, one user actually noticed this error as well. Wait, why would the delete kwarg even be generated if Python 3.12 code?

Reply:
[This is the implementation for Python 3.12:](https://github.com/python/cpython/blob/3.12/Lib/tempfile.py#L932C1-L936C58)

```python
    @classmethod
    def _cleanup(cls, name, warn_message, ignore_errors=False, delete=True):
        if delete:
            cls._rmtree(name, ignore_errors=ignore_errors)
            _warnings.warn(warn_message, ResourceWarning)
```

We use `kwargs` to avoid different versions incompatibility.

Reply:
Oh yeah that makes sense, thanks!